include/nvexec/stream/scan.cuh

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 20224 NVIDIA Corporation
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "../../stdexec/execution.hpp"
+#include <type_traits>
+#include <ranges>
+
+#include <cuda/std/type_traits>
+
+#include <cub/device/device_scan.cuh>
+
+#include "algorithm_base.cuh"
+#include "common.cuh"
+#include "../detail/throw_on_cuda_error.cuh"
+
+namespace nvexec {
+  namespace STDEXEC_STREAM_DETAIL_NS {
+    namespace scan_ {
+
+
+      template <class SenderId, class ReceiverId, class InitT, class Fun>
+      struct receiver_t
+        : public __algo_range_init_fun::receiver_t<
```

Comment:
This reuses the `algorithm_base.cuh` . The `ExclusiveScan` API I used was the one that allows you to specify an initial value, so I could easily reuse this base. 

The difference between the reduce is that it returns a single value where as a scan is to return an array of data so it is very similar.